### ISSUE TITLE
Make ews-cpp build and run for Android NDK

### ISSUE THREAD

**Issue by kmorrissey-mersive** [otris/ews-cpp#57]
We would like to use ews-cpp on a device that runs Linux/Android. Android NDK C++ libraries do not have std::to_string() and a few other functions that ews-cpp counts on. I made some minor localized changes to ews-cpp to correct for the missing functions. How do I go about submitting the changes for consideration?

**Comment by bkircher** [comment]
Hi Karen!

Android support would be a great addition and I would definitely like to merge any changes in that direction.

> How do I go about submitting the changes for consideration?

Well, just the normal "GitHub flow": Make sensible commits in a separate branch, publish that branch in your fork on GitHub, and, once you think its complete, open a [pull request](https://help.github.com/articles/about-pull-requests/).

You might want to check the [contribution guide](https://github.com/otris/ews-cpp/blob/master/CONTRIBUTING.md) for more info.

**Comment by bkircher** [comment]
Would be nice if we could support Android NDK in our CI matrix. AFAIK, [only Android SDK](https://docs.travis-ci.com/user/languages/android/) is supported in Travis CI builds.

**Comment by bkircher** [comment]
CircleCI apparently [pre-installs Android NDK](https://circleci.com/mobile/android/).

**Comment by kmorrissey-mersive** [comment]
I made a pull request with the Android NDK changes (plus a few others). I did something wrong in how I managed my repo/threads, and the pull request has some changes you should already have.

You might not want to merge yet, because when looking at the pull changes, I see some things I may have done wrong.

**Comment by kmorrissey-mersive** [comment]
One thing I was trying to fix was that on Windows VC++ under VS2012 (which we are stuck using for now) min(a,b) and max(a,b) are macros unless NOMINMAX is #defined.

I can imagine some people not wanting ews-cpp to require that NOMINMAX be defined, but I don’t know how else to do it. Do you have any thoughts?

**Comment by bkircher** [comment]
Seems like there is no libcurl for Android NDK.

**Comment by kmorrissey-mersive** [comment]
That’s not true. We use libcurl in Android NDK programs. I’m using ews-cpp right now under NDK.

**Comment by bkircher** [comment]
> That’s not true. We use libcurl in Android NDK programs. I’m using ews-cpp right now under NDK.

You're probably absolutely right. I'm totally new to Android.

**Comment by kmorrissey-mersive** [comment]
We have been running with a private, modified copy of EWS-CPP in our product. Didn't do it right the first time, so we have fallen far behind in keeping up with EWS-CPP changes. We are now setting up a proper fork and will reintegrate our Android NDK changes so we can do a usable pull request. We will be another customer for OAuth; a lot of our users use our product to access O365.

**Comment by bkircher** [comment]
Cool. Awesome news!

**Comment by idolum** [comment]
@kmorrissey-mersive Did you made progress with the Android NDK integration? Did the changes for OAuth2 work for you? I realized that there needs more to be done, if the client sits behind a proxy (see  #172).
